RegressionModels/LinearRegression/HousePriceEstimation.py

```diff
@@ -0,0 +1,174 @@
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+import numpy as np
+from sklearn.metrics import mean_squared_error
+
+#file path of the data
+file_path = "/home/panther/Downloads/train.csv"
+
+#making pandas data frame
+Data = pd.read_csv(file_path)
+
+#Identify Columns That Contain a Single Value
+Data.nunique()[Data.nunique()==1]
+
+#checking whether the coloumns contain more than 70% null values
+Data.columns[((Data.isnull().sum()/Data.shape[0])*100)>70]
+#deleting coloumns containg more than 70% of null values
+Data_removed_null = Data.drop(['Alley', 'PoolQC', 'Fence', 'MiscFeature'],axis=1)
+
+#this coloumn contains more than 90% of the single value so we are delinting this columns
+Data_removed_null['Street'].value_counts()
+Data_removed_null['Utilities'].value_counts()
+Data_removed_null['CentralAir'].value_counts()
+Data_updated = Data_removed_null.drop(['Street','Utilities','CentralAir'],axis=1)
+
+#checking for duplicate rows and deleting them
+Data_updated.drop_duplicates(inplace=True)
+
+#to find columns whcih caontaing more than 70% of single values
+for i in Data.columns:
+    print(Data[i].value_counts()[(((Data[i].value_counts())/Data.shape[0]*100)>70)])
+
+#this coloumn contains more than 70% of the single values
+Data_updated = Data_updated.drop(['MSZoning','LandContour','Condition1','BldgType','RoofStyle','RoofMatl','ExterCond','LotConfig','LandSlope','Condition2','BsmtCond','BsmtFinType2','Heating','Electrical','LowQualFinSF','BsmtHalfBath','KitchenAbvGr','Functional','GarageQual','GarageCond','PavedDrive','EnclosedPorch','3SsnPorch','PoolArea','MiscVal','SaleType','SaleCondition'],axis=1)
+
+#missing values more than 50 percentage
+num_missing = (Data_updated[Data_updated.columns] == 0).sum()
+#checking coloummns which conatin more than 50% of values
+missing_value_columns =  Data_updated.columns[num_missing>500]
+Data_updated = Data_updated.drop(missing_value_columns,axis=1)
+
+#removing coloumns which contain more null values 
+Data_updated = Data_updated.drop('FireplaceQu',axis=1)
+
```

Comment:
Create a list of columns which has

-  more than 50% null values
- single unique values
- ID based columns
and remove them in a single go using pd.drop